### DISCUSSION TITLE
VPC Reachability Analyzer

### DISCUSSION THREAD

**Discussion by kaihendry** [gruntwork-io/knowledge-base#598] [How do I]
Hi, I'm not sure terratest can help here: I am working with a complex network segregated Organisation.

I need to assert if Account A subnet 1 can communicate with Account B subnet 2.

Tbh I think AWS's VPC Reachability Analyzer is probably the best tool for the job, but I am not sure how to drive it.

Can Terratest help here?

---

<ins datetime="2022-11-24T09:25:14Z">
  <p><a href="https://support.gruntwork.io/hc/requests/109645">Tracked in ticket #109645</a></p>
</ins>

**Comment by josh-padnick** [comment]
The fundamental operating model of Terratest is:

1. Deploy a thing
2. Validate that it works as expected
3. Tear it down

VPC Reachability Analyzer seems like an excellent Terratest library candidate for step 2. I could imagine a use case where a user deploys a combination of modules such as a VPC and ECS service, and then wants to validate that the ECS service is in fact reachable. We do this naively today with a simple HTTP reach out, but if it fails, the only error is "could not reach service." The VPC Reachability Analyzer could now pinpoint exactly where the failure occurred, making debugging much easier.

PRs would be welcome here, ideally starting with the README or stubs!
